cli/azd/pkg/input/console.go

```diff
@@ -192,6 +220,39 @@ func (c *AskerConsole) MessageUxItem(ctx context.Context, item ux.UxItem) {
 	}
 }
 
+func defaultShowPreviewerOptions() *ShowPreviewerOptions {
+	return &ShowPreviewerOptions{
+		MaxLineCount: 5,
+	}
+}
+
+func (c *AskerConsole) ShowPreviewer(ctx context.Context, options *ShowPreviewerOptions) *ConsolePreviewer {
+	c.spinnerMutex.Lock()
+	defer c.spinnerMutex.Unlock()
+
+	// auto-stop any spinner
+	currentMsg := c.currentSpinnerMessage
+	c.StopSpinner(ctx, "", Step)
+
+	if options == nil {
+		options = defaultShowPreviewerOptions()
+	}
```

Comment:
The concepts of the spinner and previewer seem to be getting more interwoven together.  Showing previewer has side effects on the spinner.  Showing the spinner has side effects on the previewer.  It's getting increasingly difficult to follow.

Reply:
That's per design, and should not be hard to follow.
Components don't care about the relation between spinner/previer/etc.   The only thing which cares is the main contract, not the implementation. Implementation is replaceable.